protocol/parser/parser.go

```diff
@@ -122,102 +124,119 @@ func parseInputFromParamsWithGenericParsers(rpcInput RPCInput, genericParsers []
 	return parsed, parsedSuccessfully
 }
 
-func ParseBlockFromParams(rpcInput RPCInput, blockParser spectypes.BlockParser, genericParsers []spectypes.GenericParser) *ParsedInput {
-	parsedBlockInfo, parsedSuccessfully := parseInputFromParamsWithGenericParsers(rpcInput, genericParsers)
-	if parsedSuccessfully {
-		return parsedBlockInfo
-	}
-	if parsedBlockInfo == nil {
-		parsedBlockInfo = NewParsedInput()
+func ParseRawBlock(rpcInput RPCInput, parsedInput *ParsedInput, defaultValue string) {
+	rawBlock := parsedInput.GetBlockRaw()
+	if rawBlock == "" {
+		parsedInput.SetBlock(spectypes.NOT_APPLICABLE)
+		return
 	}
 
-	parsedBlockInfo.parsedBlock = func() int64 {
-		// first we try to parse the value with the block parser
-		result, err := parse(rpcInput, blockParser, PARSE_PARAMS)
-		if err != nil || result == nil {
-			utils.LavaFormatDebug("ParseBlockFromParams - parse failed",
-				utils.LogAttr("error", err),
-				utils.LogAttr("result", result),
-				utils.LogAttr("blockParser", blockParser),
-				utils.LogAttr("rpcInput", rpcInput),
+	parsedBlock, err := rpcInput.ParseBlock(rawBlock)
```

Comment:
add unitest to validate the fix

Reply:
Fixed in 2ac9725a327a5cf345edbebcb52d34509af3bdbe